src/components/Button/Button.tsx

```diff
@@ -179,6 +194,14 @@ class Button extends UIComponent<Extendable<IButtonProps>, any> {
       onClick(e, this.props)
     }
   }
+
+  private handleFocus = (e: React.SyntheticEvent) => {
+    const { onFocus } = this.props
+    this.setState({ isLastFocusFromKeyboard: whatInput.ask() === 'keyboard' })
+    if (onFocus) {
+      onFocus(e, this.props)
+    }
+  }
```

Comment:
In my opinion it would be useful to send the isLastFocusFromKeyboard variable to the user as well, as it is something connected to the focus. I would propose this to be restructured in the following way:
```jsx
  private handleFocus = (e: React.SyntheticEvent) => {
    const { onFocus } = this.props
    const isLastFocusFromKeyboard = whatInput.ask() === 'keyboard'
    
    if (onFocus) {
      onFocus(e, {...this.props, isLastFocusFromKeyboard})
    }

     this.setState({ isLastFocusFromKeyboard })
  }
```

Note: we need to have this additional variable as setState is asynchronous.

Reply:
would rather suggest to rename it then, as too much information is unnecessary in the state's name: something like `isFromKeyboard` should be sufficient